src/main/java/org/haffson/adventofcode/days/day01/Day01.java

```diff
@@ -41,21 +40,94 @@ public int getDay() {
 
     @Override
     public String firstPart() {
-        return "Part 1 - Frequency: " + calculateFrequency();
+        return "Part 1: " + calculatePart1();
     }
 
     @Override
     public String secondPart() {
-        return null;
+        return "Part 2: " + calculatePart2();
     }
 
     /**
      * Primary method for Day 1, Part 1.
-     * Calculates the final frequency as the sum of all frequencies.
+     * Calculates the Report
+     *
+     * @return Returns the product of the multiplication of two numbers that add up to 2020.
+     */
+    private int calculatePart1() {
+        int[] entries = this.findTwoEntriesThatSumToX(this.numbers, 2020);
+        return multiplyListEntries(entries);
+    }
+
+
+    /**
+     * Primary method for Day 1, Part 2.
+     * Calculates the Report
+     *
+     * @return Returns the product of the multiplication of three numbers that add up to 2020.
+     */
+    private int calculatePart2() {
+        int[] entries = this.findThreeEntriesThatSumToX(this.numbers, 2020);
+        return multiplyListEntries(entries);
+    }
+
+    /***
+     * This helper function calculates the product of all list entries.
+     *
+     * @param values values
+     * @return Returns the product of all list entries.
+     */
+    private int multiplyListEntries(int[] values) {
+        int result = 1;
+        for (int value : values) {
+            result = result * value;
+        }
+        return result;
+    }
+
+    /***
+     * This helper function is used to find two list entries that add up to the value x.
      *
-     * @return the final frequency
+     * @param values values
+     * @param x sum of the two list entries
+     * @return Returns the list entries witch add up to x.
      */
-    private int calculateFrequency() {
-        return 0;
+    private int[] findTwoEntriesThatSumToX(int[] values, int x) {
```

Comment:
You could turn these methods into package-private methods and test the business logic in unit tests explicitly. There you could also try to think about edge cases and whether your code covers them without running into errors.